Added export of `CPM_LAST_PACKAGE_NAME`

CraigHutchinson opened the pull request:
+ Updated docs respectively

#400

CraigHutchinson commented:
![image](https://user-images.githubusercontent.com/54269136/190109004-43993907-cceb-4379-84ee-6b6982b32826.png)
@iboB Any idea on this one?

CraigHutchinson commented:
Confirmed as working: 
![image](https://user-images.githubusercontent.com/54269136/191202271-64d9d429-c041-4e2a-8945-5fce6dd31321.png)

TheLartians commented on README.md:
```diff
@@ -73,6 +73,7 @@ After calling `CPMAddPackage`, the following variables are defined in the local
 - `<dependency>_SOURCE_DIR` is the path to the source of the dependency.
 - `<dependency>_BINARY_DIR` is the path to the build directory of the dependency.
 - `<dependency>_ADDED` is set to `YES` if the dependency has not been added before, otherwise it is set to `NO`.
+- CPM_LAST_PACKAGE_NAME is set to `<dependency>`. For single-argument syntax this will be the automatically determined name.
```
Rephrased to be more clear imo and formatted CMake identifiers as code.

```suggestion
- `CPM_LAST_PACKAGE_NAME` is set to the determined name of the last added dependency (equivalent to `<dependency>`).
```

CraigHutchinson replied:
Agreed, much better 💯

TheLartians commented on test/unit/local_dependency/ModuleCMakeLists.txt.in:
```diff
@@ -32,3 +32,4 @@ include(@CPM_PATH@/testing.cmake)
 ASSERT_TRUTHY(@TEST_DEPENDENCY_NAME@_ADDED)
 ASSERT_DEFINED(@TEST_DEPENDENCY_NAME@_SOURCE_DIR)
 ASSERT_DEFINED(@TEST_DEPENDENCY_NAME@_BINARY_DIR)
+ASSERT_EQUAL(CPM_LAST_PACKAGE_NAME "@TEST_DEPENDENCY_NAME@")
```
I believe the variable should be expanded here and the other tests below.

```suggestion
ASSERT_EQUAL("${CPM_LAST_PACKAGE_NAME}" "@TEST_DEPENDENCY_NAME@")
```

CraigHutchinson replied:
Agreed, much better 💯

CraigHutchinson replied:
Agree. Will look into these changes.

CraigHutchinson replied:
Done

TheLartians reviewed (commented):
Thanks for creating the PR and also adding tests! 

Right now the test seems a bit unfocussed, however. I think a better location for checking the correct value of `CPM_LAST_PACKAGE_NAME` might be after each `CPMAddPackage` call in the [`dependency_properties.cmake`](https://github.com/cpm-cmake/CPM.cmake/blob/10bf25a8113e2003f259536bc88ad7bc81eff8ff/test/unit/dependency_properties.cmake#L11) unit test.

TheLartians commented:
Note that you can run the unit tests locally using the following commands.

```bash
cmake -Stest -Bbuild/test
cmake --build build/test --target test-verbose
```

And also auto-format the code style style using [cmake-format](https://github.com/cheshirekow/cmake_format) as below. 

```bash
cmake -Stest/style -Bbuild/style
cmake --build build/style --target fix-format
```

CraigHutchinson commented:
> Note that you can run the unit tests locally using the following commands.

@TheLartians Thanks for this info, invaluable :)

CraigHutchinson commented:
I have updated and fixed the additional tests. However, hopefully as an aside, I get failure on this test prior & post my additional changes 
![image](https://user-images.githubusercontent.com/54269136/191457069-eca97bfb-3968-4f0c-9af1-3f491c6c2783.png)

iboB reviewed (dismissed):
I think it looks good

TheLartians reviewed (commented):
Looks good, could you run `cmake-format` or the `fix-format` target to update the code style?

TheLartians commented:
> EDIT2: It would also be useful if the UNit-test used a separate or temporary CPM_Cache location as it will currently be affected if/as there may already be an existing cache for packages of the name used in the test under some circumstances

Sounds like a good idea, though it may be even better to move all "unit-tests" that build a full project into the more extensive integration test suite.

CraigHutchinson commented:
> Looks good, could you run `cmake-format` or the `fix-format` target to update the code style?

Not made much progress here yet, getting an error and not sure how to proceed 'yet': `RuntimeError: Failed to parse C:\development\CPM.cmake\.cmake-format as any of yaml, json, or python`

iboB commented:
It seems that the style checks are still failing, though to be honest, I don't see why

TheLartians commented:
Tried it on my machine and it removed a trailing whitespace. @CraigHutchinson did you install the required dependencies? That should be easy using pip:

```bash
pip3 install clang-format==14.0.6 cmake_format==0.6.11 pyyaml
```

CraigHutchinson commented:
@TheLartians thanks for sorting the extra space, I shall try harder next time! As it stands I had issues with the fix-cmake in that it was changing line-endings in ALL the source files, I would gather in omitting this the space fix was also lost.
I haven't investigated yet, but my expectation is the issue should be git config related as the message is "LF will be replaced by CRLF the next time Git touches it"
